enhancements/20230327-harvester-seeder-integration.md

```diff
@@ -0,0 +1,223 @@
+
+* inventories.metal.harvesterhci.io
+* clusters.metal.harvesterhci.io
+* jobs.bmc.tinkerbell.org
+* machines.bmc.tinkerbell.org
+* tasks.bmc.tinkerbell.org
+
+### Implementation Overview
+
+Seeder will run as a deployment in the `harvester-system` namespace in the cluster.
+
+The addon will deploy Seeder with `EMBEDDED_MODE` set to `true`.
+
+In this mode the provisioning controllers are not bootstrapped, however the following controllers are bootstrapped
+
+* local cluster controller
+* local node contorller
+* rufio machine controller
+* rufio job controller
+* rufio task controller
+* cluster event controller
+
+On boot in embedded mode, Seeder will create a `local` cluster for `cluster.metal` objects pointing to the K8s default service as the cluster endpoint address.
+
+This is a placeholder for `inventory` objects as they are added to the cluster.
+
+```yaml
+apiVersion: metal.harvesterhci.io/v1alpha1
+kind: Cluster
+metadata:
+  creationTimestamp: "2023-03-26T22:50:15Z"
+  generation: 4
+  name: local
+  namespace: harvester-system
+  resourceVersion: "31519751"
+  uid: 44ad4855-65c7-46ba-a321-04c54ab69337
+spec:
+  clusterConfig: {}
+  nodes:
+  - addressPoolReference:
+      name: ""
+      namespace: ""
+    inventoryReference:
+      name: harvester-659jw
+      namespace: harvester-system
+  version: local
+  vipConfig:
+    addressPoolReference:
+      name: ""
+      namespace: ""
+status:
+  clusterAddress: 10.53.0.1
+  status: clusterRunning
+```
+
+The UI will allow users to create secrets and inventory definition corresponding to existing nodes.
+
+sample secret creation call to endpoint: `${ENDPOINT}/v1/harvester/secrets`
+```json
+{
+  "type": "Opaque",
+  "metadata": {
+    "namespace": "harvester-system",
+    "name": "harvester-659jw"
+  },
+  "_type": "Opaque",
+  "data": {
+    "password": "TjBJTE80dSEh",
+    "username": "RXRob3M="
+  }
+}
+```
+
+sample inventory creation call to endpoint: `${ENDPOINT}/v1/harvester/metal.harvesterhci.io.inventories`
+
+```json
+{
+  "kind": "Inventory",
+  "metadata": {
+    "name": "harvester-659jw",
+    "namespace": "harvester-system",
+    "annotations": {
+      "metal.harvesterhci.io/localInventory": "true",
+      "metal.harvesterhci.io/localNodeName": "harvester-659jw"
+    }
+  },
+  "spec": {
+    "baseboardSpec": {
+      "connection": {
+        "authSecretRef": {
+          "name": "harvester-659jw",
+          "namespace": "harvester-system"
+        },
+        "host": "172.19.1.172",
+        "insecureTLS": true,
+        "port": 623
+      }
+    },
+    "events": {
+      "enabled": true,
+      "pollingInterval": "1h"
+    }
+  }
+}
+```
+
+Since no actual provisioning is performed by Seeder, the node name the inventory is associated with is passed via the annotation `metal.harvesterhci.io/localNodeName`. The local cluster controller in seeder will use this annotation to query the k8s node object, and generate the correct inventory status.
```

Comment:
```suggestion
Since no actual provisioning is performed by Seeder, the node name the inventory is associated with is passed via the annotation `metal.harvesterhci.io/localNodeName`. The local cluster controller in Seeder will use this annotation to query the k8s node object and generate the correct inventory status.
```